### PR TITLE
Refactor AgentThreadPoolInstanceRegistry to AgentThreadPoolRegistry.

### LINKED ISSUE #1296: [AGENT] Related Refactoring of the AgentThreadPoolInstanceRegistry Class.
## Demand advice
There are two subtasks.
- [ ] Modify the Map field of AgentThreadPoolInstanceRegistry to a static field.
- [ ] Rename the class AgentThreadPoolInstanceRegistry to AgentThreadPoolRegistry for simplicity.

### PR DESCRIPTION
**Pull request by lucca-suen**
Fixes #1296 

Changes proposed in this pull request:
- Rename AgentThreadPoolInstanceRegistry to AgentThreadPoolRegistry.
- Remove the AgentThreadPoolRegistry singleton pattern design.
- Set all attributes of AgentThreadPoolRegistry that are of type Map to statically private.
- Modify how the above properties are used in other classes.

Through testing, the remote configuration center changes can be sensed in agent mode.

### CODE REVIEW

File: agent/hippo4j-agent-core/src/main/java/cn/hippo4j/agent/core/registry/AgentThreadPoolRegistry.java [hunk 1]
```diff
@@ -26,40 +26,36 @@
...
-        return holderMap;
+    public static Map<ThreadPoolExecutor, Class<?>> getEarlyConstructMap() {
+        return EARLY_CONSTRUCT_MAP;
     }
 
-    public void putHolder(String executorName, ThreadPoolExecutor executor, Properties properties) {
-        AgentThreadPoolExecutorHolder holder = new AgentThreadPoolExecutorHolder(executorName, executor, properties);
-        holderMap.put(executorName, holder);
+    public static void putEarlyConstruct(ThreadPoolExecutor executor, Class<?> earlyConstruct) {
+        EARLY_CONSTRUCT_MAP.put(executor, earlyConstruct);
```
Comment (yanrongzhen) [inline agent/hippo4j-agent-core/src/main/java/cn/hippo4j/agent/core/registry/AgentThreadPoolRegistry.java]: I think two optimizations need to be done
- No need for function wrapping via `putEarlyConstruct`
- Please rename `earlyConstruct` in the parameter list to `referencedClass`

**Review by yanrongzhen — CHANGES_REQUESTED** [review]

File: agent/hippo4j-agent-core/src/main/java/cn/hippo4j/agent/core/registry/AgentThreadPoolRegistry.java (hunk 1 above)
Comment (lucca-suen) [reply agent/hippo4j-agent-core/src/main/java/cn/hippo4j/agent/core/registry/AgentThreadPoolRegistry.java]: Good idea. I'll try to finish it later

**Comment by magestacks** [comment]
I'm refactoring this class and I'll let you know when I'm done.

**Comment by magestacks** [comment]
@lucca-suen Please synchronize the develop branch for unfinished actions.

**Comment by lucca-suen** [comment]
> @lucca-suen Please synchronize the develop branch for unfinished actions.

ok

**Pull request closed without merging**